Added "Open in New Tab" and "Open in New Window" right-click context menu options

tbrockman opened the pull request:
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

Added two context-menu options to right clicking on the paths of search results, "Open in New Tab" ~~and "Open in New Window".~~

![image](https://user-images.githubusercontent.com/8674362/28488012-f966c2aa-6e5d-11e7-84d0-9e496f7a98b3.png)

Can also open new tabs on specific line matches, and already opened tabs will becomes focused after subsequent open attempts:

![image](https://user-images.githubusercontent.com/8674362/44323027-602ea580-a40d-11e8-8f94-8902be78cdfd.png)

~~Note: Open in New Window not implemented for specific line matches as this does not appear to be easily accomplished within the current API.~~

Includes tests to guarantee that the context menu commands call the
correct functions with the intended parameters, as well as ensure that
right-clicks have no unexpected consequences with the new functionality (such as
expanding and contracting of selected results). But it should be noted that I'm still familiarizing myself with the API and tests, so it's possible that the tests may not be entirely accurate due to that lack of familiarity.

<!--

We must be able to understand the design of your change from this description. If we can't get a good idea of what the code will be doing from the description here, the pull request may be closed at the maintainers' discretion. Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code here recently, so please walk us through the concepts.

-->

### Alternate Designs

None.

### Benefits

Ability to open multiple results from directory searches without being required to navigate back to search results view. Allows for some nice ~~window and~~ tab organization and facilitates workflows that involve opening and editing multiple files matching certain search criteria.

### Possible Drawbacks

<!-- What are the possible side-effects or negative impacts of the code change? -->

People who frequently use "Copy" and "Copy Path" might accidentally click on the newly added menu options.

### Applicable Issues

N/A. This was just a feature I'd wanted myself in the past, ideally others find it useful.

50Wliu commented:
I think for this feature to be useful for me it'd need to have the context menu show up on individual results as well.

> Fixes might include forcing the tab into focus when a new tab is selected

I think this makes the most sense.

tbrockman commented:
That's fair, I'll work that into the pull request.

Now opening new tabs to a specific line is already supported by the Atom API, but do you know if there exists any sort of way to do the same with a new Atom window?

tbrockman commented:
Came back to update the code I'd written, fixed merge conflicts, updated calls to no longer existing functions, and then squashed everything into one neat commit.

If there's anything that needs to be changed or added in terms of tests or code, let me know.

lee-dohm commented:
Thanks for the help! Apologies for taking so long to get back to you.

* Please remove Open in New Window since it isn't completely implemented
* Please rename command to `find-and-replace:open-in-new-tab`

Thanks again for the contribution!

tbrockman commented:
No worries, glad to hear back!

Done and done, let me know if there's anything else.
